### PR TITLE
Add Wasm Support

### PR DESCRIPTION
**Pull request by hewigovens**
Fixes https://github.com/trustwallet/wallet-core/issues/1921, wasm binary size is around 3.7 M

### CODE REVIEW

**Review by ed738 — APPROVED** [review]

**Review by ed738 — APPROVED** [review]

**Review by ed738 — APPROVED** [review]

**Review by Stankqw — APPROVED** [review]

File: src/Ethereum/ABI/ParamFactory.h [hunk 1]
```diff
@@ -14,6 +14,8 @@
 #include <string>
 #include <vector>
 
+#include "Wasm.h"
```
Comment (catenocrypt) [inline src/Ethereum/ABI/ParamFactory.h:17]: Is this really needed here?

**Review by catenocrypt — APPROVED** [review]

File: src/Ethereum/ABI/ParamFactory.h (hunk 1 above)
Comment (hewigovens) [reply src/Ethereum/ABI/ParamFactory.h:17]: It fails to build without some int defines

File: tools/wasm-build [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -e
+
+base_dir=$(cd `dirname $0`; pwd)
+src_dir=${base_dir}/..
+build_folder=${src_dir}/wasm-build
+boost_dir=${EMSDK}/upstream/emscripten/system/include
```
Comment (catenocrypt) [inline tools/wasm-build:8]: The variable EMSDK was not set on my system, after running the install-wasm-dependencies script.  Setting is manually worked.

File: samples/wasm/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+> Use it at your own risk.
+
+## Prerequisites
+
+1. Everything you need to build the wallet core: https://developer.trustwallet.com/wallet-core/developing-the-library/building
+2. Install [emsdk](https://emscripten.org/docs/getting_started/downloads.html)
+- - python3
+- - cmake
+- - run `tools/install-wasm-dependencies`
+
```
Comment (catenocrypt) [inline samples/wasm/README.md:16]: Maybe add a step to check for variable EMSDK, and if not set, set it.

Comment (hewigovens) [reply samples/wasm/README.md:16]: good point, updated readme

**Pull request merged**